src/main/java/io/jenkins/plugins/fontawesome/FontAwesomeIcons.java

```diff
@@ -0,0 +1,110 @@
+package io.jenkins.plugins.fontawesome;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.*;
+import java.util.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Stream;
+
+/**
+ * Utility to work with icons provided by the font-awesome-api-plugin.
+ *
+ * @author strangelookingnerd
+ */
+public final class FontAwesomeIcons {
+
+    private static final Logger LOGGER = Logger.getLogger(FontAwesomeIcons.class.getName());
+    private static final String SVG_FILE_ENDING = ".svg";
+    private static final String IMAGES_SYMBOLS_PATH = "images/symbols/";
+    private static final String FONT_AWESOME_API_PLUGIN = "font-awesome-api";
+    private static final String ICON_CLASS_NAME_PATTERN = "symbol-%s plugin-" + FONT_AWESOME_API_PLUGIN;
+
+    private FontAwesomeIcons() {
+        // hidden
+    }
+
+    private static Map<String, String> getIconsFromClasspath(final SvgTag.FontAwesomeStyle filter) {
+        try {
+            Enumeration<URL> urls = FontAwesomeIcons.class.getClassLoader().getResources(IMAGES_SYMBOLS_PATH);
+
+            while (urls.hasMoreElements()) {
+                URL url = urls.nextElement();
+
+                if (StringUtils.contains(url.toExternalForm(), FONT_AWESOME_API_PLUGIN)) {
+                    URI uri = url.toURI();
+
+                    if (StringUtils.equals(uri.getScheme(), "jar")) {
+                        try (FileSystem fileSystem = FileSystems.newFileSystem(uri, Collections.emptyMap())) {
+                            return collectIcons(fileSystem.getPath(IMAGES_SYMBOLS_PATH), filter);
+                        }
+                    }
+                    else {
+                        return collectIcons(Paths.get(uri), filter);
+                    }
+                }
+            }
+        }
+        catch (IOException | URISyntaxException ex) {
+            LOGGER.log(Level.WARNING, "Unable to read available icons: Resource unavailable.", ex);
+        }
+
+        return new LinkedHashMap<>();
+    }
+
+    private static Map<String, String> collectIcons(final Path path, final SvgTag.FontAwesomeStyle filter) throws IOException {
+        Map<String, String> icons = new LinkedHashMap<>();
+
+        if (path != null) {
+            try (Stream<Path> stream = Files.walk(filter == null ? path : path.resolve(filter.name().toLowerCase(Locale.ENGLISH)), filter == null ? 2 : 1)) {
+                stream.filter(icon -> icon != null && icon.getFileName() != null && StringUtils.endsWith(icon.getFileName().toString(), SVG_FILE_ENDING))
+                        .sorted().forEach(icon -> {
```

Comment:
I think here you need to add a suppression.

Reply:
Okay, I finally got to the root of the warning (https://github.com/spotbugs/spotbugs/issues/724).
A simple suppression does not do the trick for lambdas, moving the code into separate methods is ... well ... ugly to say the least.

Please let me know how to proceed here.